### ISSUE TITLE
Full Text Search not working in Virtuoso 7

### ISSUE THREAD

**Issue by Rahien** [openlink/virtuoso-opensource#107]
Hi,

In the latest development version of virtuoso 7 (commit d2117b6e08518c1ed7fc41b191969723dfa66c55) I cannot get the sparql full text indexing to work. I followed the instructions on http://docs.openlinksw.com/virtuoso/sparqlextensions.html.

I tried:
DB.DBA.RDF_OBJ_FT_RULE_ADD (null, null, 'All');
DB.DBA.VT_INC_INDEX_DB_DBA_RDF_OBJ ();
DB.DBA.VT_INDEX_DB_DBA_RDF_OBJ ();

But my search select \* where {?s ?p ?o. ?o bif:contains "'test'".} does not return any results. This search does work however: select \* where {?s ?p ?o. FILTER (regex(?o,"test*"))}
It returns, amongst others, http://lod2.tenforce.com/wp8/dataset/hhosIcUoXTvuzxiGkKUNAuzccbFQWZOy   http://purl.org/dc/terms/title  "test one"@en.

I would very much like to use the actual free text indexing though.

Interestingly:
SQL> DB.DBA.RDF_OBJ_FT_RECOVER () ;
STRG
VARCHAR

---

Temporary drop of rule "All" for graph <> predicate <>...
... done
Restoring rule "All" for graph <> predicate <>...
... done

**\* Error 22023: [Virtuoso Driver][Virtuoso Server]SR008: Function sprintf needs an integer as argument 2, not an arg of type VARCHAR (182)
at line 5 of Top-Level:
DB.DBA.RDF_OBJ_FT_RECOVER ()

**Comment by openlink** [comment]
In VOS 7, you do not need to perform these commands as this index is created already at startup.

To make sure all your data is indexed, please run the following command:

```
  DB.DBA.VT_INC_INDEX_DB_DBA_RDF_OBJ () 
```

which should check if there are any triples, that need reindexing and make sure the results are stored in the index.

I suggest you upgrade your VOS7 to the latest version from the develop/7 branch which is going to be released in the next couple of days

**Comment by flamingofugang** [comment]
Hi,
Following this issue, I am using the latest Virtuoso. Without running any stored procedures after installation, the 'bif:contains' works fine and very fast. However, in my case, I want to do exact string match case-insensitive, whereas 'bif:contains' does substring match. I have searched for other bif functions, but the documentations are few. I found the following function may be related: bif:strstr; bif:strcasestr, bif:strcontains. I have tried all of these, but they are very slow. 

My question is whether there is a way to do exact string match much faster than sparql filter regex in virtuoso7?
